pkg/template/api/validation/validation.go

```diff
@@ -51,3 +55,116 @@ func validateTemplateBody(template *api.Template) (allErrs field.ErrorList) {
 	allErrs = append(allErrs, unversionedvalidation.ValidateLabels(template.ObjectLabels, field.NewPath("labels"))...)
 	return
 }
+
+// ValidateTemplate tests if required fields in the Template are set.
+func ValidateTemplateInstance(oc *client.Client, ctx kapi.Context, ti *api.TemplateInstance) (allErrs field.ErrorList) {
```

Comment:
@bparees this isn't checking if the impersonated user can instantiate arbitrary objects from the template (indeed, that is already done as you state).  This is checking if the user creating the TemplateInstance object with a given ImpersonteUser has permissions to impersonate that user.  I can move this to an admission plugin.